proto/agent.go

```diff
@@ -219,16 +216,22 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 	switch detail.Status {
 	case evergreen.TaskSucceeded:
 		tc.logger.Task().Info("Task completed - SUCCESS.")
+		grip.Info("Running post task commands")
+		a.runPostTaskCommands(ctx, tc)
+		grip.Info("Finished running post task commands")
 	case evergreen.TaskFailed:
 		tc.logger.Task().Info("Task completed - FAILURE.")
+		grip.Info("Running post task commands")
+		a.runPostTaskCommands(ctx, tc)
+		grip.Info("Finished running post task commands")
 	case evergreen.TaskUndispatched:
 		tc.logger.Task().Info("Task completed - ABORTED.")
+	case evergreen.TaskConflict:
+		tc.logger.Task().Error("Task completed - 409 FROM API SERVER")
+		// If we receive a 409, return control to the loop (ask for a new task)
+		return nil, nil
```

Comment:
I would call this message "Task completed - CANCELED."

Reply:
Done.